MapboxNavigation/RouteMapViewController.swift

```diff
@@ -924,6 +930,7 @@ protocol RouteMapViewControllerDelegate: NavigationMapViewDelegate, MGLMapViewDe
     func mapViewControllerDidCancelNavigation(_ mapViewController: RouteMapViewController)
     func mapViewController(_ mapViewController: RouteMapViewController, didSend feedbackId: String, feedbackType: FeedbackType)
     func mapViewControllerShouldAnnotateSpokenInstructions(_ routeMapViewController: RouteMapViewController) -> Bool
+    func mapViewController(_ mapViewController: RouteMapViewController, roadNameAt location: CLLocation) -> String?
```

Comment:
@1ec5 wouldn't you say, it would be much safer to return a `nil`. Anything else should be assigned -- hence we don't have to even check for an empty string. I think it's much elegant that way. 
(An empty string will imply the developer actually intended to set an empty string for unknown street names). Thoughts?

Reply:
@1ec5 Sure. Why don't we maintain a consistent function return type of `nil` and within `labelCurrentRoad(at:)` we check for `roadName` like so `delegate?.mapViewController(self, roadNameAt: location) ?? ""`. 
And then we can still do a check for `roadName.isEmpty`. Same approach but a different point of evaluation. 😺

Reply:
https://github.com/mapbox/mapbox-navigation-ios/pull/1309/files#diff-292af689ce88d5b5aa06966456abc0e9R690

@1ec5 ☝️

Reply:
We need to distinguish between the following three cases:

* `NavigationViewControllerDelegate.navigationViewController(_:roadNameAt:)` is implemented and returns a road name: show the label and don’t fall back to the tile-querying implementation.
* `NavigationViewControllerDelegate.navigationViewController(_:roadNameAt:)` is implemented but says there’s no road name: hide the label and don’t fall back to the tile-querying implementation.
* `NavigationViewControllerDelegate.navigationViewController(_:roadNameAt:)` is unimplemented: fall back to the tile-querying implementation.

Reply:
Got it! Makes sense.

Reply:
@1ec5 done ✅